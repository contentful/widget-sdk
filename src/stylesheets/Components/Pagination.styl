.pagination
  display: flex
  justify-content: space-between
  margin-top: 20px

.pagination__nav
  display: flex
  align-items: baseline

.pagination__text
  color: $colors.text-light

.pagination__button,
.pagination__limit-selector
  margin-left: 20px;
