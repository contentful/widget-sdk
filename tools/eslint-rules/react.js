module.exports = {
  extends: ['plugin:react/recommended', 'prettier/react'],
  plugins: ['react-hooks'],
  settings: {
    react: {
      version: '16.10.2',
    },
  },
  rules: {
    'react-hooks/rules-of-hooks': 'error',
    'react-hooks/exhaustive-deps': 'error',
    'react/display-name': 'off',
    'react/prop-types': ['error', { ignore: ['children'] }],
    'react/no-unused-prop-types': 'error',
    'react/jsx-no-target-blank': ['error', { enforceDynamicLinks: 'always' }],
  },
};
