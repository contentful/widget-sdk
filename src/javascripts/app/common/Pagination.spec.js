import React from 'react';

import { render, fireEvent } from '@testing-library/react';

import Pagination from './Pagination';

describe('Pagination', () => {
  const onChangeFn = jest.fn();

  const mount = (skip, limit, total, loading) => {
    return render(
      <Pagination skip={skip} limit={limit} total={total} loading={loading} onChange={onChangeFn} />
    );
  };

  it('disables the Previous button if it is the first page', () => {
    const { getByTestId } = mount(0, 10, 100, false);
    const previous = getByTestId('pagination.previous');
    expect(previous).toBeDisabled();
  });

  it('enables the Previous and Next buttons if it is not the first page or last pages', () => {
    const { getByTestId } = mount(10, 10, 100, false);
    const previous = getByTestId('pagination.previous');
    const next = getByTestId('pagination.next');
    expect(previous).not.toBeDisabled();
    expect(next).not.toBeDisabled();
  });

  it('disables the Next button if it is the last page', () => {
    const { getByTestId } = mount(90, 10, 100, false);
    const next = getByTestId('pagination.next');
    expect(next).toBeDisabled();
  });

  it('disables the Next and Previous buttons there is only one avialable page', () => {
    const { getByTestId } = mount(0, 10, 10, false);
    const previous = getByTestId('pagination.previous');
    const next = getByTestId('pagination.next');
    expect(previous).toBeDisabled();
    expect(next).toBeDisabled();
  });

  it('disables the Next and Previous buttons there is loading', () => {
    const { getByTestId } = mount(20, 10, 100, true);
    const previous = getByTestId('pagination.previous');
    const next = getByTestId('pagination.next');
    expect(previous).toBeDisabled();
    expect(next).toBeDisabled();
  });

  it('triggers onChange on the Next button click', () => {
    const currentSkip = 10;
    const limit = 10;
    const desiredSkip = currentSkip + limit;
    const { getByTestId } = mount(currentSkip, limit, 100, false);
    const next = getByTestId('pagination.next');
    fireEvent.click(next);
    expect(onChangeFn).toHaveBeenCalledWith({ skip: desiredSkip, limit });
  });

  it('triggers onChange on the Previous button click', () => {
    const currentSkip = 10;
    const limit = 10;
    const desiredSkip = currentSkip - limit;
    const { getByTestId } = mount(currentSkip, limit, 100, false);
    const previous = getByTestId('pagination.previous');
    fireEvent.click(previous);
    expect(onChangeFn).toHaveBeenCalledWith({ skip: desiredSkip, limit });
  });

  it('trigger onChange on the limit selector change', () => {
    const { getByTestId } = mount(10, 10, 100, false);
    const limitSelector = getByTestId('pagination.limit');
    fireEvent.change(limitSelector, { target: { value: 25 } });
    expect(onChangeFn).toHaveBeenCalledWith({ skip: 10, limit: 25 });
  });
});
