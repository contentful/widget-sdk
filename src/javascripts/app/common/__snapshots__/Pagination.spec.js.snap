// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination disables the Next and Previous buttons there is loading 1`] = `
<div
  className="pagination"
>
  <nav
    className="pagination__nav"
  >
    <p
      className="pagination__text"
    >
      View
    </p>
    <Select
      className="pagination__limit-selector"
      hasError={false}
      isDisabled={false}
      onChange={[Function]}
      required={false}
      testId="cf-ui-select"
      value="10"
      width="small"
    >
      <Option
        value="10"
      >
        10
      </Option>
      <Option
        value="25"
      >
        25
      </Option>
      <Option
        value="50"
      >
        50
      </Option>
      <Option
        value="100"
      >
        100
      </Option>
    </Select>
  </nav>
  <nav
    className="pagination__nav"
  >
    <p
      className="pagination__text"
    >
      Showing 
      21 - 30
       of 
      100
    </p>
     
    <Button
      buttonType="muted"
      className="pagination__button"
      disabled={true}
      indicateDropdown={false}
      isFullWidth={false}
      loading={false}
      onClick={[Function]}
      testId="cf-ui-button"
      type="button"
    >
      Previous
    </Button>
     
    <Button
      buttonType="muted"
      className="pagination__button"
      disabled={true}
      indicateDropdown={false}
      isFullWidth={false}
      loading={false}
      onClick={[Function]}
      testId="cf-ui-button"
      type="button"
    >
      Next
    </Button>
  </nav>
</div>
`;

exports[`Pagination disables the Next and Previous buttons there is only one avialable page 1`] = `
<div
  className="pagination"
>
  <nav
    className="pagination__nav"
  >
    <p
      className="pagination__text"
    >
      View
    </p>
    <Select
      className="pagination__limit-selector"
      hasError={false}
      isDisabled={false}
      onChange={[Function]}
      required={false}
      testId="cf-ui-select"
      value="10"
      width="small"
    >
      <Option
        value="10"
      >
        10
      </Option>
      <Option
        value="25"
      >
        25
      </Option>
      <Option
        value="50"
      >
        50
      </Option>
      <Option
        value="100"
      >
        100
      </Option>
    </Select>
  </nav>
  <nav
    className="pagination__nav"
  >
    <p
      className="pagination__text"
    >
      Showing 
      1 - 10
       of 
      10
    </p>
     
    <Button
      buttonType="muted"
      className="pagination__button"
      disabled={true}
      indicateDropdown={false}
      isFullWidth={false}
      loading={false}
      onClick={[Function]}
      testId="cf-ui-button"
      type="button"
    >
      Previous
    </Button>
     
    <Button
      buttonType="muted"
      className="pagination__button"
      disabled={true}
      indicateDropdown={false}
      isFullWidth={false}
      loading={false}
      onClick={[Function]}
      testId="cf-ui-button"
      type="button"
    >
      Next
    </Button>
  </nav>
</div>
`;

exports[`Pagination disables the Next button if it is the last page 1`] = `
<div
  className="pagination"
>
  <nav
    className="pagination__nav"
  >
    <p
      className="pagination__text"
    >
      View
    </p>
    <Select
      className="pagination__limit-selector"
      hasError={false}
      isDisabled={false}
      onChange={[Function]}
      required={false}
      testId="cf-ui-select"
      value="10"
      width="small"
    >
      <Option
        value="10"
      >
        10
      </Option>
      <Option
        value="25"
      >
        25
      </Option>
      <Option
        value="50"
      >
        50
      </Option>
      <Option
        value="100"
      >
        100
      </Option>
    </Select>
  </nav>
  <nav
    className="pagination__nav"
  >
    <p
      className="pagination__text"
    >
      Showing 
      91 - 100
       of 
      100
    </p>
     
    <Button
      buttonType="muted"
      className="pagination__button"
      disabled={false}
      indicateDropdown={false}
      isFullWidth={false}
      loading={false}
      onClick={[Function]}
      testId="cf-ui-button"
      type="button"
    >
      Previous
    </Button>
     
    <Button
      buttonType="muted"
      className="pagination__button"
      disabled={true}
      indicateDropdown={false}
      isFullWidth={false}
      loading={false}
      onClick={[Function]}
      testId="cf-ui-button"
      type="button"
    >
      Next
    </Button>
  </nav>
</div>
`;

exports[`Pagination disables the Previous button if it is the first page 1`] = `
<div
  className="pagination"
>
  <nav
    className="pagination__nav"
  >
    <p
      className="pagination__text"
    >
      View
    </p>
    <Select
      className="pagination__limit-selector"
      hasError={false}
      isDisabled={false}
      onChange={[Function]}
      required={false}
      testId="cf-ui-select"
      value="10"
      width="small"
    >
      <Option
        value="10"
      >
        10
      </Option>
      <Option
        value="25"
      >
        25
      </Option>
      <Option
        value="50"
      >
        50
      </Option>
      <Option
        value="100"
      >
        100
      </Option>
    </Select>
  </nav>
  <nav
    className="pagination__nav"
  >
    <p
      className="pagination__text"
    >
      Showing 
      1 - 10
       of 
      100
    </p>
     
    <Button
      buttonType="muted"
      className="pagination__button"
      disabled={true}
      indicateDropdown={false}
      isFullWidth={false}
      loading={false}
      onClick={[Function]}
      testId="cf-ui-button"
      type="button"
    >
      Previous
    </Button>
     
    <Button
      buttonType="muted"
      className="pagination__button"
      disabled={false}
      indicateDropdown={false}
      isFullWidth={false}
      loading={false}
      onClick={[Function]}
      testId="cf-ui-button"
      type="button"
    >
      Next
    </Button>
  </nav>
</div>
`;

exports[`Pagination enables the Previous and Next buttons if it is not the first page or last pages 1`] = `
<div
  className="pagination"
>
  <nav
    className="pagination__nav"
  >
    <p
      className="pagination__text"
    >
      View
    </p>
    <Select
      className="pagination__limit-selector"
      hasError={false}
      isDisabled={false}
      onChange={[Function]}
      required={false}
      testId="cf-ui-select"
      value="10"
      width="small"
    >
      <Option
        value="10"
      >
        10
      </Option>
      <Option
        value="25"
      >
        25
      </Option>
      <Option
        value="50"
      >
        50
      </Option>
      <Option
        value="100"
      >
        100
      </Option>
    </Select>
  </nav>
  <nav
    className="pagination__nav"
  >
    <p
      className="pagination__text"
    >
      Showing 
      11 - 20
       of 
      100
    </p>
     
    <Button
      buttonType="muted"
      className="pagination__button"
      disabled={false}
      indicateDropdown={false}
      isFullWidth={false}
      loading={false}
      onClick={[Function]}
      testId="cf-ui-button"
      type="button"
    >
      Previous
    </Button>
     
    <Button
      buttonType="muted"
      className="pagination__button"
      disabled={false}
      indicateDropdown={false}
      isFullWidth={false}
      loading={false}
      onClick={[Function]}
      testId="cf-ui-button"
      type="button"
    >
      Next
    </Button>
  </nav>
</div>
`;
